### DISCUSSION TITLE
How to enable high fps?

### DISCUSSION THREAD

**Discussion by eecoder-dyf** [rustdesk/rustdesk#6324] [General]
I run RustDesk  on Linux (as server) and  Windows  (as host) in my home with 10Gbps LAN
The codec is set as H.265 (with NVIDIA RTX3080Ti on server and NVIDIA GTX1650 on host). 
The monitor resoulution is 3840×2160.
However, the control fps is always limited to 15 fps, no matter how I adjust the fps in settings.

**Comment by 21pages** [comment]
try https://github.com/rustdesk/rustdesk/discussions/5264 on windows

**Reply by eecoder-dyf** [reply]
I tried this, but it just improved from 15fps to 20fps. The gpu encoder is not fully used (just 20%)

**Reply by 21pages** [reply]
what if you control yourself, GTX1650 controls GTX1650 ? linux doesn't support new gpu acceleration, the slow encoding will reduce the fps.

**Reply by eecoder-dyf** [reply]
Still limted to 20 fps when I try to control windows by windows

**Reply by 21pages** [reply]
The fps will only increase when the picture is dynamic.

**Reply by eecoder-dyf** [reply]
![屏幕截图 2023-11-08 222530](https://github.com/rustdesk/rustdesk/assets/59913015/c6239489-8234-4681-8636-430f5305e680)
The scene is playing a 60 fps video, it can only reach 20 fps

**Reply by 21pages** [reply]
kvm-win11, It's  virtual machine? did you disable old hardware codec?

**Reply by eecoder-dyf** [reply]
It's a virtual machine exactly with GPU-passthrough. Do you mean that I should disable "use hardware codec" and only enable "enable zero copy gpu acceleration" in the settings? But if I do so, I can't enable H.264 and H.265 in remote control.

**Reply by 21pages** [reply]
> disable "use hardware codec" and only enable "enable zero copy gpu acceleration" in the settings

Yes, `hardware codec` is old implement, if you enable it and the new one is not supported, it will fallback to the old.

If you have windows host, not virtual machine, its ID is 1234, just connect to 1234 itself for test, new zero copy gpu acceleration is not designed for virtual machine. BTW, virtual machine can use the old implement for encoding, I don't know that before.

If still not supported with both side host (not vm), please show the file content of `%appdata%\RustDesk\config\RustDesk_gpucodec.toml` if you didn't install it.

**Reply by eecoder-dyf** [reply]
It exactly works when I try both host, thanks! And will you try to support Linux later? There are a lot of remote control apps on Windows, but still lack of remote control apps with high fps on Linux.

**Reply by 21pages** [reply]
Thanks for the test, linux will be  supported after windows is stable. Could you click upvote at https://github.com/rustdesk/rustdesk/discussions/5264#discussioncomment-7214549 and https://github.com/rustdesk/rustdesk/discussions/5264#discussioncomment-7231048, thanks.
